src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -236,7 +238,17 @@ private void SolutionManager_ActionsExecuted(object sender, ActionsExecutedEvent
                     if (e.Actions.Any(action =>
                         NuGetProject.GetUniqueNameOrName(action.Project) == projectName))
                     {
-                        Refresh();
+                        // With .NET Core projects, we need to make sure we only refresh once,
+                        // since the ActionsExecuted and CacheUpdated likely both report the same update
+                        if (DateTimeOffset.Now.Subtract(_lastProjectUpdateRefreshExecuted).TotalMilliseconds > 2000)
+                        {
+                            if (_lastProjectUpdateRefreshExecuted.Equals(DateTimeOffset.MinValue))
+                            {
+                                _lastProjectUpdateRefreshExecuted = DateTimeOffset.Now;
```

Comment:
If the intention is to only refresh once, it could be done with a bool, rather than DateTimeOffset. If the intention is to refresh at most once per 2 seconds, I think there's a bug as `_lastProjectUpdateRefreshExecuted ` can only be updated once and 2 seconds later, regardless of how quickly events happen, it will refresh every event.